### PR TITLE
INT B-20928

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

Security wanted us to not use `MD5` hashing, so transitioning to `SHA256` for uploads and adding in the `CheckSumAlgorithem` when uploading to S3.

This also contains code to fix some flaky tests. Idk, man.

### How to test

1. You'll need to go to your `.envrc` and uncomment `export STORAGE_BACKEND=s3`
2. Run `direnv allow`
3. Run `aws-vault exec transcom-gov-dev -- make server_run`
4. While that's building, make sure you are in the right AWS profile - `transcom-gov-dev`
6. This allows us to upload files to S3 instead of locally ([docs here](https://dp3.atlassian.net/wiki/spaces/MT/pages/1470955567/How+to+test+storing+data+in+S3+locally))
7. Log in as a customer, fill out all information and upload some orders
8. In your CLI, run `aws-vault login transcom-gov-dev` - this will open the AWS Console
9. Search for S3 and get in there
10. Find your buckets
11. There should be one that says `transcom-gov-dev-app-devlocal-us-gov-west-1`, click it
12. You should see your name/repo if the file uploaded correctly (ex. `daniel.jordan_cn`)
13. Click through and click the file you uploaded to where you see the `Object Overview` section
14. Scroll down and you should see an `Additional checksums` section
15. Confirm you see an additional `SHA-256` checksum

## Screenshots
Screenshot 2024-10-29 at 11 15 28 AM

### CODE REVIEW

**Review by lucyluuuu — APPROVED** [review]
Did not see new vulnerabilities on files changed when scanned locally. Pre-test jobs ran successfully.

File: pkg/storage/storage.go [hunk 1]
```diff
@@ -43,22 +33,11 @@ type FileStorer interface {
...
-	//RA: This line was flagged because of the use of MD5 hashing
-	//RA: This line of code hashes the AWS object to be able to verify data integrity
-	//RA: Purpose of this hash is to protect against environmental risks, it does not
-	//RA: hash any sensitive user provided information such as passwords
-	//RA: AWS S3 API requires use of MD5 to validate data integrity.
-	//RA Developer Status: Mitigated
-	//RA Validator Status: Mitigated
-	//RA Validator: jneuner@mitre.org
-	//RA Modified Severity: CAT III
-	// #nosec G401
```
Comment (deandreJones) [inline pkg/storage/storage.go:60]: I find it interesting that truss had this finding suppressed

File: pkg/storage/storage.go [hunk 2]
```diff
@@ -43,22 +33,11 @@ type FileStorer interface {
...
+// ComputeChecksum calculates the SHA256 checksum for the provided data. It expects that
 // the passed io object will be seeked to its beginning and will seek back to the
 // beginning after reading its content.
 func ComputeChecksum(data io.ReadSeeker) (string, error) {
-	//RA Summary: gosec - G401 - Weak cryptographic hash
-	//RA: This line was flagged because of the use of MD5 hashing
-	//RA: This line of code hashes the AWS object to be able to verify data integrity
-	//RA: Purpose of this hash is to protect against environmental risks, it does not
-	//RA: hash any sensitive user provided information such as passwords
-	//RA: AWS S3 API requires use of MD5 to validate data integrity.
```
Comment (deandreJones) [inline pkg/storage/storage.go:55]: is this statement true "AWS S3 API requires use of MD5 to validate data integrity." have we validated that?

File: pkg/storage/storage.go (hunk 1 above)
Comment (danieljordan-caci) [reply pkg/storage/storage.go:60]: Def thought it was weird when I saw it, too

File: pkg/storage/storage.go (hunk 2 above)
Comment (danieljordan-caci) [reply pkg/storage/storage.go:55]: It does not anymore. We can now pass in
```
ChecksumAlgorithm:    types.ChecksumAlgorithmSha256,
```

into the S3 struct, as you can see in the changes for `pkg/storage/s3.go`

**Review by deandreJones — APPROVED** [review]

**Pull request merged**